### ISSUE TITLE
internal/reports/reports.go:917:23: nilness: nil dereference in index operation (govet)

### ISSUE THREAD

**Issue by atc0005** [atc0005/check-statuspage#86]
Surfaced by unstable container linting settings while validating #85. 

From the `govet` `nilness` analyzer:

```
internal/reports/reports.go:917:23: nilness: nil dereference in index operation (govet)
		for i, err := range errsEncountered {
		                    ^
```

I've seen this before, but believed it was a false-positive match at the time:

https://github.com/atc0005/check-statuspage/blob/0cf6d60c77072cc82a8466ead8854d2b0a5d76bd/internal/reports/reports.go#L909-L926

I saw that the potential nil dereference was guarded against by a length check here (this code is responsible for populating the slice, it isn't received from another function):

https://github.com/atc0005/check-statuspage/blob/0cf6d60c77072cc82a8466ead8854d2b0a5d76bd/internal/reports/reports.go#L909

Looking closer, I think I likely copy/pasted/modified this function when initially developing the project and left the above behind:

https://github.com/atc0005/check-statuspage/blob/0cf6d60c77072cc82a8466ead8854d2b0a5d76bd/internal/reports/reports.go#L435-L441

In short, I think I can yank L909-L926 as it's not needed here.
